src/user-lists/user-list.service.ts

```diff
@@ -309,7 +309,16 @@ export class UserListService {
     queryBuilder.orderBy("full_name", orderBy);
     queryBuilder.offset(pageOptionsDto.skip).limit(pageOptionsDto.limit);
 
-    const [itemCount, entities] = await Promise.all([queryBuilder.getCount(), queryBuilder.getMany()]);
+    const subQuery = this.userHighlightRepository.manager
+      .createQueryBuilder()
+      .from(`(${queryBuilder.getQuery()})`, "subquery_for_count")
+      .setParameters(queryBuilder.getParameters())
+      .select("count(full_name)");
+
+    const countQueryResult = await subQuery.getRawOne<{ count: number }>();
+    const itemCount = parseInt(`${countQueryResult?.count ?? "0"}`, 10);
+
+    const [entities] = await Promise.all([queryBuilder.getRawMany()]);
```

Comment:
This doesn't need a `Promise.all` anymore since there's only one promise:

```suggestion
    const entities = await queryBuilder.getRawMany();
```